workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowRunsTabContent.tsx

```diff
@@ -88,25 +109,86 @@ export const WorkflowRunsTabContent = () => {
             field: 'state',
           }
         : undefined;
-    const workflowIdFilter: FieldFilter | undefined = workflowId
-      ? {
-          operator: 'EQ',
-          value: workflowId,
-          field: 'processId',
+
+    let startedFilter: FieldFilter | undefined = undefined;
+
+    if (startedSelectorValue !== Selector.AllItems) {
+      let dateRange: [string, string] | undefined = undefined;
+
+      const currentDate = new Date();
+      const endOfToday = new Date(currentDate);
+      endOfToday.setHours(23, 59, 59, 999);
+
+      switch (startedSelectorValue) {
+        case 'Today': {
+          const startOfToday = new Date();
+          startOfToday.setHours(0, 0, 0, 0);
+          dateRange = [startOfToday.toISOString(), endOfToday.toISOString()];
+          break;
         }
-      : undefined;
-    if (statusFilter && workflowIdFilter) {
+        case 'Yesterday': {
+          const startOfYesterday = new Date();
+          startOfYesterday.setDate(startOfYesterday.getDate() - 1);
+          startOfYesterday.setHours(0, 0, 0, 0);
+
+          const endOfYesterday = new Date(startOfYesterday);
+          endOfYesterday.setHours(23, 59, 59, 999);
+
+          dateRange = [
+            startOfYesterday.toISOString(),
+            endOfYesterday.toISOString(),
+          ];
+          break;
+        }
+        case 'Last 7 days': {
+          const startOfLast7Days = new Date();
+          startOfLast7Days.setDate(startOfLast7Days.getDate() - 7);
+          startOfLast7Days.setHours(0, 0, 0, 0);
+
+          dateRange = [
+            startOfLast7Days.toISOString(),
+            endOfToday.toISOString(),
+          ];
+          break;
+        }
+        case 'This month': {
+          const startOfCurrentMonth = new Date();
+          startOfCurrentMonth.setDate(1);
+          startOfCurrentMonth.setHours(0, 0, 0, 0);
+
+          dateRange = [
+            startOfCurrentMonth.toISOString(),
+            endOfToday.toISOString(),
+          ];
+          break;
+        }
+        default:
+          dateRange = undefined;
+      }
+
+      startedFilter =
+        startedSelectorValue !== Selector.AllItems
+          ? {
+              operator: 'BETWEEN',
+              value: dateRange,
+              field: 'start',
+            }
+          : undefined;
+    }
+
+    // removes undefined filters
+    const filters = [statusFilter, workflowIdFilter, startedFilter].filter(
+      Boolean,
+    ) as FieldFilter[];
+
+    if (filters.length > 1) {
       return {
         operator: 'AND',
-        filters: [statusFilter, workflowIdFilter],
+        filters,
       };
-    } else if (statusFilter) {
-      return statusFilter;
-    } else if (workflowIdFilter) {
-      return workflowIdFilter;
     }
-    return undefined;
-  }, [statusSelectorValue, workflowId]);
+    return filters[0] || undefined;
```

Comment:
why `filters[0]`? can't be there more than one valid filter?

Reply:
This looks good to me since `if (filters.length > 1)` above handles that case